pypeit/core/mosaic.py

```diff
@@ -0,0 +1,312 @@
+        Users should appropriately consider how these pixels will affect the
+        mosaic pixels *before* calling this function.
+
+        Similarly, error propagation from the input image to the mosaic image is
+        only approximate when ``order > 0``.  Error propagaion is performed
+        simply by applying the coordinate transform to each variance image with
+        the same order as used for the input image, and then combining those
+        variances as necessary in overlap regions.
+
+        Tests show that this approach is also not invertable.  I.e., iteratively transforming the image back and forth between the native and mosaic frames lead to image drifts.
+
+    Args:
+        imgs (:obj:`list`, `numpy.ndarray`_):
+            List of `numpy.ndarray`_ images to include in the mosaic.  The shape
+            of all the input images must be identical if ``mosaic_shape`` is
+            None.
+        tforms (:obj:`list`, `numpy.ndarray`_):
+            List of `numpy.ndarray`_ objects with the transformation matrices
+            necessary to convert between image and mosaic coordinates.  See
+            :func:`pypeit.core.mosaic.build_image_mosaic_transform`.  The number
+            of transforms must match the number of images.  If ``mosaic_shape``
+            is None, the transforms are considered in a relative sense.  That
+            is, the shape of the output mosaic is determined by applying these
+            transforms to the bounding boxes of each image and then determining
+            the shape needed to retain all pixels in the input images.  The
+            transforms are then adjusted appropriately to map to this shape; see
+            :func:`~pypeit.core.mosaic.prepare_mosaic`.  If ``mosaic_shape`` is
+            *not* None, these transforms are expected to map directly to the
+            output mosaic coordinates.
+        ivar (:obj:`list`, `numpy.ndarray`_, optional):
+            List of `numpy.ndarray`_ images with the inverse variance of the
+            image data.  The number of inverse-variance images must match the
+            number of images in the mosaic.  If None, inverse variance is
+            returned as None.
+        bpm (:obj:`list`, `numpy.ndarray`_, optional):
+            List of boolean `numpy.ndarray`_ objects with the bad-pixel mask for
+            each image in the mosaic.  The number of bad-pixel masks must match
+            the number of images in the mosaic.  If None, all input pixels are
+            considered valid.
+        mosaic_shape (:obj:`tuple`, optional):
+            Shape for the output image.  If None, the shape is determined by
+            :func:`pypeit.core.mosaic.prepare_mosaic` and the shape of all the
+            input images *must* be identical.
+        cval (:obj:`float`, optional):
+            The value used to fill empty pixels in the mosaic.
+        order (:obj:`int`, optional):
```

Comment:
Can you for convenience provide information on how order is defined. I realise that scipy.ndimage.affine_transform has that documented, but even there you have to dig to figure out that order =0 is NGP and order = 1 is linear.

Reply:
Done.